Navigation/include/ActsLUXEPipeline/LUXEMeasurementsCreator.hpp

```diff
@@ -0,0 +1,246 @@
+#pragma once
+
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/EventData/Measurement.hpp"
+#include "Acts/EventData/detail/TestSourceLink.hpp"
+#include "ActsLUXEPipeline/LUXESimpleSourceLink.hpp"
+#include "ActsLUXEPipeline/IAlgorithm.hpp"
+#include "ActsLUXEPipeline/LUXEEffectiveMaterial.hpp"
+#include "ActsLUXEPipeline/LUXEMeasurement.hpp"
+
+#include "ActsLUXEPipeline/DataHandle.hpp"
+
+#include "ActsLUXEPipeline/LUXEDataContainers.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Geometry/GeometryHierarchyMap.hpp"
+#include "Acts/MagneticField/MagneticFieldContext.hpp"
+#include "Acts/Propagator/EigenStepper.hpp"
+#include "Acts/Propagator/AbortList.hpp"
+#include "Acts/Propagator/ActionList.hpp"
+#include "Acts/Propagator/StandardAborters.hpp"
+#include "Acts/Utilities/Logger.hpp"
+#include "Acts/Navigation/DetectorNavigator.hpp"
+#include "ActsFatras/Physics/ElectroMagnetic/Scattering.hpp"
+#include "ActsFatras/Physics/ElectroMagnetic/BetheBloch.hpp"
+#include "ActsFatras/Physics/ElectroMagnetic/BetheHeitler.hpp"
+
+
+#include <memory>
+#include <random>
+#include <vector>
+#include <iostream>
+#include <fstream>
+
+#include <boost/archive/binary_oarchive.hpp>
+#include <boost/archive/binary_iarchive.hpp>
+
+/// Propagator action to create measurements.
+namespace LUXENavigator {
+
+    using namespace Acts::UnitLiterals;
+
+    struct MeasurementsCreatorAction {
+
+        using result_type = LUXEDataContainer::SimMeasurements;
+        std::size_t sourceId = 0;
+        /// @brief Operator that is callable by an ActionList. The function
+        /// collects the surfaces
+        ///
+        /// @tparam propagator_state_t Type of the propagator state
+        /// @tparam stepper_t Type of the stepper
+        /// @tparam navigator_t Type of the navigator
+        ///
+        /// @param [out] result Vector of matching surfaces
+        /// @param [in] state State of the propagator
+        template<typename propagator_state_t, typename stepper_t,
+                typename navigator_t>
+        void operator()(propagator_state_t &state, const stepper_t &stepper,
+                        const navigator_t &navigator, result_type &result,
+                        const Acts::Logger &logger) const {
+            using namespace Acts::UnitLiterals;
+            Acts::Vector4 trueVertex;
+            if (state.steps == 0) {
+                trueVertex = {stepper.position(state.stepping)[0],
+                              stepper.position(state.stepping)[1],
+                              stepper.position(state.stepping)[2],
+                              stepper.time(state.stepping)};
+            }
+            if (state.steps > 100) {
```

Comment:
You can set `maxSteps` when you initialize `Navigator`/`Propagator`.

Reply:
But then it just raises an error instead of going to the next event